### PR TITLE
Add `CARGO_MAKE_WORKSPACE_INCLUDE_MEMBERS`.

### LINKED ISSUE #315: Filter for workspace member
**Features Description**
Ability to define list of workspace members to be used and all others to be skipped.

**Describe the solution you'd like**
An environment variable like `CARGO_MAKE_WORKSPACE_FILTER_MEMBERS` to select specific members to be used and none other. Orthogonal to `CARGO_MAKE_WORKSPACE_SKIP_MEMBERS`.

**Code Sample**
```toml
[env]
CARGO_MAKE_WORKSPACE_FILTER_MEMBERS = "member3;member4"
```

### PR DESCRIPTION
**Pull request by daxpedda**
I think I managed to add a working test this time, might be more rigorous though.
The documentation has to be regenerated, didn't manage to get that working.

Fixes #315.

### CODE REVIEW

File: src/lib/execution_plan.rs [hunk 1]
```diff
@@ -120,6 +120,26 @@ fn should_skip_workspace_member(member: &str, skipped_members: &HashSet<String>)
     }
 }
 
+fn get_included_workspace_members(include_members_config: String) -> Option<HashSet<String>> {
+    let list = get_skipped_workspace_members(include_members_config);
```
Comment (sagiegurari) [inline src/lib/execution_plan.rs]: this is a bit unreadable to me.
maybe we can have a single function like is_workspace_member_found(member, members_map) -> bool
and then should_skip_workspace_member calls it and if found returns skip and should_include_workspace_member calls it and if found returns include.
same logic should also be for get_skipped_workspace_members and get_included_workspace_members.

File: src/lib/execution_plan_test.rs [hunk 2]
```diff
@@ -315,6 +315,54 @@ cd -"#
     assert!(task.env.is_none());
 }
 
+#[test]
+#[cfg(target_os = "linux")]
+fn create_workspace_task_with_included_members() {
```
Comment (sagiegurari) [inline src/lib/execution_plan_test.rs]: can we also have a test that tests the combination of both include and skip?

File: src/lib/execution_plan_test.rs [hunk 3]
```diff
@@ -315,6 +315,54 @@ cd -"#
...
+        "member1".to_string(),
+        "member2".to_string(),
+        "dir1/member3".to_string(),
+    ];
+    crate_info.workspace = Some(Workspace {
+        members: Some(members),
+        exclude: None,
+    });
+
+    envmnt::remove("CARGO_MAKE_USE_WORKSPACE_PROFILE");
```
Comment (sagiegurari) [inline src/lib/execution_plan_test.rs]: we are not testing profile switch, so we can remove this one and just put 'development' hardcoded in the line

File: src/lib/execution_plan_test.rs [hunk 4]
```diff
@@ -315,6 +315,54 @@ cd -"#
...
+cd ./member2
+cargo make --disable-check-for-updates --allow-private --no-on-error --loglevel=LEVEL_NAME --env CARGO_MAKE_CRATE_CURRENT_WORKSPACE_MEMBER=member2 --profile PROFILE_NAME some_task
+cd -"#
+        .to_string();
+
+    let log_level = logger::get_log_level();
+    expected_script = str::replace(&expected_script, "LEVEL_NAME", &log_level);
+
+    let profile_name = profile::get();
+    expected_script = str::replace(&expected_script, "PROFILE_NAME", &profile_name);
```
Comment (sagiegurari) [inline src/lib/execution_plan_test.rs]: same comment about the profile. we can drop this

File: README.md [hunk 5]
```diff
@@ -1686,6 +1686,29 @@ You will have to invoke this as a composite flow:
...
+```
+
+You can also define glob paths, for example:
+
+```toml
+[env]
+CARGO_MAKE_WORKSPACE_INCLUDE_MEMBERS = "tools/*"
+```
+
+This works together with **CARGO_MAKE_WORKSPACE_SKIP_MEMBERS**, members who are skipped will not be included.
```
Comment (sagiegurari) [inline README.md]: This works together with **CARGO_MAKE_WORKSPACE_SKIP_MEMBERS**, members which are excluded.

File: README.md [hunk 6]
```diff
@@ -1686,6 +1686,29 @@ You will have to invoke this as a composite flow:
 cargo make workspace-task --no-workspace
 ```
 
+<a name="usage-workspace-support-include-members"></a>
+#### Include Specific Members
```
Comment (sagiegurari) [inline README.md]: great. can you also add a link to this in the nav.md?

File: src/lib/execution_plan.rs (hunk 1 above)
Comment (daxpedda) [reply src/lib/execution_plan.rs]: Done. Did I get it right?

File: src/lib/execution_plan_test.rs (hunk 2 above)
Comment (daxpedda) [reply src/lib/execution_plan_test.rs]: Done.

File: README.md (hunk 6 above)
Comment (daxpedda) [reply README.md]: Done.

File: README.md (hunk 5 above)
Comment (daxpedda) [reply README.md]: I tried to include that **including** members doesn't not ignore rules set by **skipping** members. Please feel free to modify it.

> This works together with **CARGO_MAKE_WORKSPACE_SKIP_MEMBERS**, which will still exclude members and not be overwritten by **CARGO_MAKE_WORKSPACE_INCLUDE_MEMBERS**.

**Comment by sagiegurari** [comment]
looks great. thanks a lot for all your PRs.
keep in mind that currently 0.24 release is pretty much complete, so i'll keep it in development for few days to get some feedback or maybe last moment requests before publishing an official release.
if you do have more items you would like to push, please open new issues today/tomorrow.

**Comment by daxpedda** [comment]
Thanks for creating and maintaining this great project!
> if you do have more items you would like to push, please open new issues today/tomorrow.

I will, thanks.

**Pull request merged**